python/tests/rtl/test_api_settings.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=redefined-outer-name
+import configparser
+import pytest
+
+from looker.rtl import api_settings
+
+
+@pytest.fixture(scope='session')
+def config_file(tmpdir_factory):
+    """Creates a sample looker.ini file and returns its path"""
+    filename = tmpdir_factory.mktemp('settings').join('looker.ini')
+    filename.write("""
+[Looker]
+# API version is required
+api_version=3.1
+# Base URL for API. Do not include /api/* in the url
+base_url=https://host1.looker.com:19999
+# API 3 client id
+client_id=your_API3_client_id
+# API 3 client secret
+client_secret=your_API3_client_secret
+# Optional embed secret for SSO embedding
+embed_secret=your_embed_SSO_secret
+# Optional user_id to impersonate
+user_id=
+# Set to false if testing locally against self-signed certs. Otherwise leave True
+verify_sl=True
+# leave verbose off by default
+verbose=false
+
+[Looker2]
+api_version=3.0
+base_url=https://host2.looker.com:19999
+client_id=your_API3_client_id
+client_secret=your_API3_client_secret
+embed_secret=your_embed_SSO_secret
+user_id=
+verify_ssl=True
+
+[Looker3]
+base_url=https://host3.looker.com:19999/
+client_id=myclientid
+client_secret=myclientsecret
+        """)
+    return filename
+
+
+def test_settings_defaults_to_looker_section(config_file):
+    """ApiSettings should retrieve settings from default (Looker) section 
+    if section is not specified during instantiation."""
+    settings = api_settings.ApiSettings(config_file)
+    assert settings.base_url == 'https://host1.looker.com:19999'
+
+
+@pytest.mark.parametrize("test_section, expected_url",
```

Comment:
👍 for parametrize. though having spent some time in other language testing frameworks I find the configuration of `parametrize` to be a bit kludgey. I wish you could pass in a list of `NamedTuple` (or even better `@dataclass`) instances and use the attribute names as the test func args. The "list as a string" construct here hurts my eyes a little :-)